app/controllers/users_controller.rb

```diff
@@ -4,6 +4,7 @@ def new
 
   def create
     if User.find_by(new_email)
+      duplicate_email
```

Comment:
I guess this is fine, but I wonder if in real life it's more common for people to try to re-sign up for something or for them to incorrectly type their own email and put in someone else's? Maybe the latter, which means that your logic is the way to go.